#356 Tag changes ...again

Rev219 opened the pull request:
For some reason I couldn't update the previous branch at all, so I made a new one and applied the new tags here.

The new tags are divided into 2 main categories: Weapons and Tools, with the hybrid being an edge case for axes. The weapon and tool tag are applied directly to the main_hand_slot tag, and the hybrid tag is applied to the weapon and tool tag respectively.

ImplementsLegend commented on src/main/java/com/wanderersoftherift/wotr/init/WotrTags.java:
```diff
@@ -32,6 +32,11 @@ public static class Items {
+        public static final TagKey<Item> SOCKETABLE_MAIN_HAND_SLOT_WEAPON = createTag(
+                "socketable_main_hand_slot_weapon");
+        public static final TagKey<Item> SOCKETABLE_MAIN_HAND_SLOT_TOOL = createTag("socketable_main_hand_slot_tool");
+        public static final TagKey<Item> SOCKETABLE_MAIN_HAND_SLOT_HYBRID = createTag(
+                "socketable_main_hand_slot_hybrid");
```
I mean it's used in the datagen, but shouldn't be used anywhere in the actual logic. But feel free to ignore this, it was an unimportant nitpick anyway.

Rev219 commented:
> Runegem update where?

Should I try and add new modifiers that only apply to tools/weapons respectively?

ImplementsLegend commented:
I'd say updating current ones to be weapons-only is enough. (Or does someone want those on tools as well?)

WarrenGore commented:
I'm fine with you leaving them as they are if they still function fine. There will be more adjustments to the runegem/modifier distribution in the coming weeks so don't need to get too caught up in it here. Main goal we were accomplishing is opening up the ability to provide tool modifiers separately from weapon modifiers.

However, if you really want to do something with them I would update ones that only make sense on weapons to be weapons only (for example: damage, life leech, attack speed). And maybe a sample one for tools (I think you had mining efficiency) so there is something to look at for anyone adding more afterwards. Anything else we can review as we get a closer look into runegem shape/modifier to socketed item distribution.

Rev219 commented:
Ok, added an example mining runegem that uses both the tool tag and the hybrid tag. Also changed some of the weapon related runegems to have the mainHandWeaponTag tag.

immortius commented on src/main/java/com/wanderersoftherift/wotr/datagen/WotrRuneGemDataProvider.java:
```diff
@@ -582,6 +588,17 @@ public static void bootstrapRuneGems(BootstrapContext<RunegemData> context) {
+                                )), new RunegemData.ModifierGroup(
+                                        mainHandHybridTag, List.of(
+                                                new TieredModifier(1, getModifier(lookup, "mining_speed")),
+                                                new TieredModifier(1, getModifier(lookup, "attack_flat"))
```
Is this supposed to have attack_flat?
